File: benchmarks/programs/primecount/primecount.go
package main

import "fmt"

func main() {
	count := 0
	for n := 2; n <= 10000; n++ {
		isPrime := true
		for d := 2; d*d <= n; d++ {
			if n%d == 0 {
				isPrime = false
				break
			}
		}
		if isPrime {
			count++
		}
	}
	fmt.Println(count)
}

File: benchmarks/programs/primecount/primecount.rs
fn main() {
    let mut count = 0;
    for n in 2..=10000 {
        let mut is_prime = true;
        let mut d = 2;
        while d * d <= n {
            if n % d == 0 {
                is_prime = false;
                break;
            }
            d += 1;
        }
        if is_prime {
            count += 1;
        }
    }
    println!("{}", count);
}
